src/commands.rs

```diff
@@ -206,11 +204,13 @@ fn redirect_stderr(f: File) -> io::Result<()> {
 }
 
 /// If `SCCACHE_ERROR_LOG` is set, redirect stderr to it.
-fn redirect_error_log() -> io::Result<()> {
-    match env::var("SCCACHE_ERROR_LOG") {
-        Ok(filename) => OpenOptions::new().create(true).append(true).open(filename).and_then(redirect_stderr),
-        _ => Ok(()),
-    }
+fn redirect_error_log() -> Result<()> {
+    let name = match env::var("SCCACHE_ERROR_LOG") {
+        Ok(filename) => filename,
+        _ => return Ok(()),
+    };
+    let f = OpenOptions::new().create(true).append(true).open(name)?;
+    redirect_stderr(f)
```

Comment:
We apparently have inherently different styles about using intermediate variables. :) I think I err on the "way too long functional programming" side of things. Probably all that LISP programming in my formative years!